apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -87,18 +87,20 @@ private function register_patterns() {
 			register_block_pattern_category( $slug, array( 'label' => $label ) );
 		}
 
-		foreach ( (array) $this->get_patterns() as $pattern ) {
+		foreach ( (array) $block_patterns as $pattern ) {
 			if ( $this->can_register_pattern( $pattern ) ) {
+				$is_premium = in_array( 'premium_block_pattern', array_keys( $pattern['tags'] ), true );
```

Comment:
Once D53825-code lands, we'll just need to update this line to check for `$pattern['pattern_meta']['is_premium']`.